deploy/kubernetes/ai-gateway/inference-pool/inference-pool.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: inference.networking.x-k8s.io/v1alpha2
+kind: InferencePool
```

Comment:
This use of InferencePool does not seem to match with the Gateway API Inference SIG's design intent for InferencePool. I do see some reasoning why you chose to do this.  However this can potentially cause issues down the road. Is there a design document where you provide some background for this design choice ?